src/Stratis.Bitcoin/Base/BackgroundWork/IAsyncDelegateWorker.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace Stratis.Bitcoin.Base.BackgroundWork
+{
+    public interface IAsyncDelegateWorker : IDisposable { }
+
+    /// <summary>
+    /// Define a delegate that is called asynchronously in the background whenever a new <typeparamref name="T"/> is queued and run it.
+    /// </summary>
+    /// <typeparam name="T">Type of the queued items used in the delegate.</typeparam>
+    /// <seealso cref="System.IDisposable" />
+    public interface IAsyncDelegateWorker<T> : IAsyncDelegateWorker { }
```

Comment:
This compont is both not needed and will intervene when splitting AsyncQueue to its two componentes `AsyncDelegates` and `AsyncDequeuer` I suggest just return the interface when creating the respective Async[Component] and thats enough (they anyway have thier own creation method)

Reply:
I'm going to split AsyncQueu right after this PR push (or within this PR if we want all here)
TBH I'm not even sure the delegate creation should return anything

>>>Probably we may not even need a return value when creating an async delegate because they are cancelled using a CancellationTokenSource.

Reply:
yeah true, the delegate creation needs to return only am IDIsposable probably.

Also ok you can split AsynqQueue later but at least split the interfaces now.